scripts/debug_tools/failure_lib.py

```diff
@@ -0,0 +1,53 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+import json
+import os
+
+
+def findPathEnd(string, path_begin):
+    """ Find one path in string.
+    Returns the pair of the begin and end pos.
+    """
+    path_end = string.find(' ', path_begin)
+    if path_end == -1:
+        path_end = len(string)
+    path = string[path_begin:path_end]
+    return path, path_end
+
+
+def changePaths(string, pathModifierFun):
+    """ Scan through the string and possibly replace all found paths.
+    Returns the modified string.
+    """
+    result = ''
+    i = 0
+    while i < len(string):
+        # Everything is a path wich starts with a '/' and there is a whitespace
+        # after that.
+        # Note, this supports only POSIX paths.
+        if string[i] == '/':
+            path, path_end = findPathEnd(string, i)
+            path = pathModifierFun(path)
+            result += path
+            i = path_end - 1
+        else:
+            result += string[i]
+        i = i + 1
+    return result
+
+
+class IncludePathModifier:
+    def __init__(self, sources_root):
+        self.sources_root = sources_root
+
+    def __call__(self, path):
+        return os.path.join(self.sources_root, path.lstrip(os.path.sep))
+
+
+def load_json_file(filename):
```

Comment:
Do we really want a utility function for this code snippet?
If so, do we want it here?

What if the file opening fails? Where should it be handled?

Reply:
My goal is to handle the file opening, if the file is not in place, the default OSError exception is perfectly fine here.